### ISSUE TITLE
Make plebbit options more similar to plebbit-js plebbit options

### ISSUE THREAD

**Issue by estebanabaroa** [plebbit/plebbit-cli#6]
The current API is:

```
FLAGS
  --ipfsApiPort=<value>      (required) [default: 32429] Specify the API port of the ipfs node to listen on
  --ipfsGatewayPort=<value>  (required) [default: 32430] Specify the gateway port of the ipfs node to listen on
  --plebbitApiPort=<value>   (required) [default: 32431] Specify Plebbit API port to listen on
  --plebbitDataPath=<value>  (required) [default: /home/runner/.local/share/plebbit] Path to plebbit data path where
```

Wouldn't it be better if the API was more consistent with plebbit-js. ie:

```
  --ipfsGatewayUrl
  --ipfsHttpClientOptions
  --pubsubHttpClientOptions
  --dataPath
```

I would make them singuar, but allow passing more than 1. e.g.:
```
plebbit --ipfsGatewayUrl https://ipfs.io --ipfsGatewayUrl https://cloudflare-ipfs.com
```
For chainProviders we could handle multiple URLs with:
```
plebbit --chainProviders.eth.url site.com --chainProviders.eth.url site2.com
```

I'm not sure how we would handle ipfsHttpClientOptions as options, maybe we dont allow it, or maybe we do:

```
plebbit --ipfsHttpClientOptions.0.url https://site1.com --ipfsHttpClientOptions.1.url https://site2.com
```
or

```
plebbit --ipfsHttpClientOptions[0].url https://site1.com --ipfsHttpClientOptions[1].url https://site2.com
```

**Comment by Rinse12** [comment]
Are you referring to the command `plebbit daemon`? The assumption is the daemon will boot up its own IPFS node and the args `ipfsApiPort` and `ipfsGatewayPort` will be used to set the ports the IPFS API and gateway will listen on. It's not for connecting to other IPFS APIs or gateways.

There should be a way to set chainProviders for the daemon though. Maybe something like `plebbit daemon --chainProviders.eth[0].url https://ethrpc.com`

`--plebbitDataPath` and `plebbitApiPort` should be changed to `--dataPath` and `plebbitRpcApiPort`

**Comment by Rinse12** [comment]
- imo all plebbit-js options should go under `plebbitOptions`. We change `--plebbitDataPath` to `--plebbitOptions.plebbitDataPath`, and allow other plebbit options like chainProviders to be set like `--plebbitOptions.chainProviders.eth.url https://ethgateway.com`.

- Replace port flags with url flags, instead of `--ipfsApiPort 12345`, it would be `--ipfsApiUrl http://localhost:12345`. That would allow users to set a remote ipfs node url instead of having to boot up a new ipfs node, which the current implementation forces. 

We should allow setting array values via flags without having to declare the flag multiple times, ie `plebbit daemon --plebbitOptions.ipfsGatewayUrls 'http://gateway1.com' 'https://gateway2.com'`

**Comment by estebanabaroa** [comment]
> We should allow setting array values via flags without having to declare the flag multiple times, ie `plebbit daemon --plebbitOptions.ipfsGatewayUrls 'http://gateway1.com' 'https://gateway2.com'`

I like having `plebbitOptions.something` instead of just `something` but I don't think an array can be done. I dont know any cli programs that do this. they only allow multiple options when it's the last argument, for example `rm file1 file2 file3`. I dont know any CLI program that lets you do `command --flag something something something --flag something something something`, they usually force you to repeat them, and I think repeating is fine. IMO we dont care that it's long to write, people will only set it up once and probably prepare the command in a text file or something.

**Comment by estebanabaroa** [comment]
> * Replace port flags with url flags, instead of `--ipfsApiPort 12345`, it would be `--ipfsApiUrl http://localhost:12345`. That would allow users to set a remote ipfs node url instead of having to boot up a new ipfs node, which the current implementation forces.

well isnt ipfsApiPort used by the plebbit-cli process to launch kubo? if you launch kubo, you're always on localhost. if you want to use a remote kubo node, why not just set --plebbitOptions.ipfsHttpClientsOption "http://remotekubonode"?

**Comment by Rinse12** [comment]
>if you launch kubo, you're always on localhost.

A user may want kubo to listen on `0.0.0.0` or other network interfaces (like docker) instead of `127.0.0.1` like it does by default now. imo `--ipfsApiUrl` should be flexible to accommodate as many setups as possible.

>if you want to use a remote kubo node, why not just set --plebbitOptions.ipfsHttpClientsOption "http://remotekubonode/"?

I agree we should be doing that. Although if a user provides both `--plebbitOptions.ipfsHttpClientsOptions` and `--ipfsApiUrl` it should throw. So I guess `--ipfsApiUrl` will be an optional value that defaults to `http://localhost:5001` when `plebbitOptions.ipfsHttpClientsOptions` is undefined. Same thing for `--ipfsGatewayPort`

- for `--plebbitApiPort`, it should be changed to `--plebbitRpcApiUrl` with `ws://localhost:9138` as a default, for the same reasons we changed `--ipfsApiPort` above. `--plebbitRpcApiUrl` should always be required, and  `--plebbitOptions.plebbitRpcClientOptions` should be forbidden because there's no point in starting a local daemon just to connect to another daemon

**Comment by estebanabaroa** [comment]
> A user may want kubo to listen on `0.0.0.0` or other network interfaces (like docker) instead of `127.0.0.1` like it does by default now. imo `--ipfsApiUrl` should be flexible to accommodate as many setups as possible.

true I didnt think about that, since kubo supports putting a full IP in the config, we should also support it so that the settings are just passed from plebbit-cli to kubo directly

**Comment by estebanabaroa** [comment]
> if a user provides both `--plebbitOptions.ipfsHttpClientsOptions` and `--ipfsApiUrl` it should throw. So I guess `--ipfsApiUrl` will be an optional value that defaults to `http://localhost:5001` when `plebbitOptions.ipfsHttpClientsOptions` is undefined.

seems fine

**Comment by estebanabaroa** [comment]
> * for `--plebbitApiPort`, it should be changed to `--plebbitRpcApiUrl` with `ws://localhost:9138` as a default, for the same reasons we changed `--ipfsApiPort` above. ``

we can't use RpcApi imo, it has to be API or RPC, not both. and since we already called it plebbit rpc everywhere, and that it's a websocket json rpc, not a rest api like kubo, we should probably call it plebbitRpcUrl, and since rpc-websockets accepts a "host" param, we could use pass the plebbitRpcUrl to the host field, though this could be buggy, like it might block requests that use name.com as "host" even though the user meant to accept them, so maybe we should not use the "host" param yet.

**Comment by Rinse12** [comment]
Should be implemented from version 0.15.0 and on
